lib/xml/encrypt.js

```diff
@@ -28,24 +28,29 @@ exports.unencrypted = function (xml, callback) {
 
 exports.encrypted = function (encryptOptions) {
   return function encrypt(xml, callback) {
-    xmlenc.encrypt(xml, encryptOptions, function (err, encrypted) {
-      if (err) {
-        // Attempt to fix errors and retry
-        console.warn(`Failed to encrypt xml. Attempting to fix errors and retrying. Error=${err.message}`)
-        xmlenc.encrypt(xml,
-            {
-                ...encryptOptions,
-                rsa_pub: utils.fixPemFormatting(encryptOptions.rsa_pub),
-                pem: utils.fixPemFormatting(options.encryptionCert)
-            },
-            function (retryErr, retryEncrypted) {
-              if (retryErr) return callback(retryErr)
-              return callback(null, utils.removeWhitespace(retryEncrypted));
-        })
-      }
-
-      if (err) return callback(err)
-      return callback(null, utils.removeWhitespace(encrypted));
-    });
+    try {
```

Comment:
try/catch only works for code that is synchronous or uses async/await. If xmlenc.encrypt is synchronous, this will work, but similar to a comment last week, the callback continuation style implies an asynchronous operation. If it's synchronous in reality, that's an implementation detail we shouldn't rely on since the API contract states otherwise.

I think what we want here is something like:

```js
xmlenc.encrypt(xml, encryptOptions, function (err, encrypted) {
  if (err) {
    // Attempt to fix errors and retry
    console.warn(
      `Failed to encrypt xml. Attempting to fix errors and retrying. Error=${err.message}`
    );
    xmlenc.encrypt(
      xml,
      {
        ...encryptOptions,
        rsa_pub: utils.fixPemFormatting(encryptOptions.rsa_pub),
        pem: utils.fixPemFormatting(options.encryptionCert),
      },
      function (retryErr, retryEncrypted) {
        if (retryErr) {
          return callback(retryErr);
        }

        callback(null, utils.removeWhitespace(retryEncrypted));
      }
    );
  } else {
    callback(null, utils.removeWhitespace(encrypted));
  }
});

```

Reply:
Somehow I forgot about that quirk of async calls. I borrowed your code.